src/upp/upp.py

```diff
@@ -65,32 +66,45 @@ def _get_pp_data_from_registry(reg_file_path):
 
     return tmp_pp_file.name
 
+def _check_file_writeable(filename):
+    if os.path.exists(filename):
+        if os.path.isfile(filename):
+            return os.access(filename, os.W_OK)
+        else:
+            return False
+    pdir = os.path.dirname(filename)
+    if not pdir: pdir = '.'
+    return os.access(pdir, os.W_OK)
 
 def _write_pp_to_reg_file(filename, data, debug=False):
-    reg_string = REG_KEY_VAL[3:] + '"=hex:' + data.hex(',')
-    reg_lines = [reg_string[i:i+75] for i in range(0, len(reg_string), 75)]
-    reg_lines[0] = '"' + REG_KEY_VAL[:3] + reg_lines[0]
-    formatted_reg_string = '\\\r\n  '.join(reg_lines)
-    reg_pp_data = REG_HEADER + formatted_reg_string + 2 * '\r\n'
-    if debug:
-        print(reg_pp_data)
-    decode._write_pp_tables_file(filename, reg_pp_data.encode('utf-16'))
-    print('Written {} Soft PowerPlay bytes to {}'.format(len(data), filename))
-
+    if _check_file_writeable(filename):
+        reg_string = REG_KEY_VAL[3:] + '"=hex:' + data.hex(',')
+        reg_lines = [reg_string[i:i+75] for i in range(0, len(reg_string), 75)]
+        reg_lines[0] = '"' + REG_KEY_VAL[:3] + reg_lines[0]
+        formatted_reg_string = '\\\r\n  '.join(reg_lines)
+        reg_pp_data = REG_HEADER + formatted_reg_string + 2 * '\r\n'
+        if debug:
+            print(reg_pp_data)
+        decode._write_pp_tables_file(filename, reg_pp_data.encode('utf-16'))
+        print('Written {} Soft PowerPlay bytes to {}'.format(len(data), filename))
+    else: 
+        print('Can not write to {}'.format(filename))
     return 0
 
 @click.group(context_settings=CONTEXT_SETTINGS)
 @click.option('-p', '--pp-file', help='Input/output PP table binary file.',
               metavar='<filename>',
               default='/sys/class/drm/card0/device/pp_table')
+@click.option('-t', '--to-registry', help='Output to Windows registry .reg file.',
+              metavar='<filename>')
```

Comment:
Since this option only makes sense for **set** command, can you move the decorator back there?